disco_aws_automation/resource_helper.py

```diff
@@ -105,15 +101,13 @@ def throttled_call(fun, *args, **kwargs):
             if (error_code not in ("Throttling", "RequestLimitExceeded")) or (time.time() > expire_time):
                 raise
 
-            time.sleep(curr_delay)
-            delay_register = last_delay
-            last_delay = curr_delay
-            curr_delay = min(curr_delay + delay_register, MAX_POLL_INTERVAL)
+            cycle = backoff(cycle)
 
 
 def wait_for_state(resource, state, timeout=15 * 60, state_attr='state'):
     """Wait for an AWS resource to reach a specified state"""
     time_passed = 0
```

Comment:
It doesn't look like we are ever updating`time_passed`. Are we using this (we probably should be)